proto/xudrpc.proto

```diff
@@ -527,22 +527,26 @@ message SwapResult {
   double quantity = 4 [json_name = "quantity"];
   // The hex-encoded payment hash for the swaps.
   string r_hash = 5 [json_name = "r_hash"];
+  // The hex-encoded preimage.
+  string r_preimage = 6 [json_name = "r_preimage"];
+  // The price of the order that's being executed.
+  int64 price = 7 [json_name = "r_preimage"];
   // The amount of subunits (satoshis) received.
-  int64 amount_received = 6 [json_name = "amount_received"];
+  int64 amount_received = 8 [json_name = "amount_received"];
```

Comment:
I think mainly for the sake of minimizing unnecessary code changes we shouldn't change the number of existing fields here and just add the preimage and price at the end as 12 and 13. Changing the field numbers can actually break backwards compatibility with consumers of the service.